src/components/Input/Input.tsx

```diff
@@ -151,3 +151,56 @@ export const Input: FC<Props> = ({
     </div>
   )
 }
+
+interface NewInputProps extends Props {
+  classes?: string
+  inputClasses?: string
+  restInputProps?: JSX.IntrinsicElements['input'] & NumericFormatProps<InputAttributes>
+}
+
+/**
+ * New input from the new design system May 2025
+ * @param type
+ * @param classes
+ * @param inputClasses
+ * @param placeholder
+ * @param value
+ * @param onChange
+ * @param name
+ * @param readonly
+ * @param restInputProps
+ * @constructor
+ */
+export const NewInput = ({
+  type,
+  classes,
+  inputClasses,
+  placeholder,
+  value,
+  onChange,
+  name,
+  readonly,
+  restInputProps,
+}: NewInputProps) => {
+  switch (type) {
+    case 'number':
+      return (
+        <InputNumber
+          className={cn(classes, inputClasses)}
+          placeholder={placeholder}
+          value={value}
+          onValueChange={({ value }) => onChange?.(value)}
+          name={name}
+          data-testid={`Input_${name}`}
```

Comment:
Avoid `undefined` by marking `name` as required or assigning it a default value like an empty string.

Reply:
Name is a required prop, this will never be undefined unless the developer makes a mistake which will be caught by TS